Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -96,4 +96,9 @@ export class ArtifactEngine {
 
     private artifactItemStore: ArtifactItemStore = new ArtifactItemStore();
     private logger: Logger;
-}
\ No newline at end of file
+}
+
+process.on('unhandledRejection', (reason, promise) => {
```

Comment:
what is this fix for?

Reply:
if the code fails at some point in node like de-referencing undefined object it generates an unhandledRejection warning, we need to map it to error to fail in such cases.